input/otlp/metadata_test.go

```diff
@@ -335,13 +337,148 @@ func TestResourceLabels(t *testing.T) {
+	}
+
+	otlp.TranslateSpan(ptrace.SpanKindInternal, m, e)
+
+	assert.Equal(t, "v", e.GetLabels()["s"].Value)
+	assert.Equal(t, "true", e.GetLabels()["b"].Value)
+	assert.Equal(t, float64(1), e.GetNumericLabels()["i"].Value)
+	assert.Equal(t, float64(1), e.GetNumericLabels()["f"].Value)
+}
+
+func TestTranslateSpanSlice(t *testing.T) {
+	const key = "k"
+	testCases := []struct {
+		input                 func() pcommon.Map
+		desc                  string
+		expectedLabel         []string
+		expectedNumericLabels []float64
+		expectedLen           int
+	}{
+		{
+			desc: "drop non-string elements",
+			input: func() pcommon.Map {
+				m := pcommon.NewMap()
+				s := m.PutEmptySlice(key)
+				s.FromRaw([]any{"", "a", 1, "b", 2})
+				return m
+			},
+			expectedLen:   3,
+			expectedLabel: []string{"", "a", "b"}, // Explicitly check that empty string is stored
+		},
+		{
+			desc: "drop non-bool elements",
+			input: func() pcommon.Map {
+				m := pcommon.NewMap()
+				s := m.PutEmptySlice("k")
+				s.FromRaw([]any{true, "a", false, 1.1})
+				return m
+			},
+			expectedLen:   2,
+			expectedLabel: []string{"true", "false"},
+		},
+		{
+			desc: "drop non-int elements",
+			input: func() pcommon.Map {
+				m := pcommon.NewMap()
+				s := m.PutEmptySlice("k")
+				s.FromRaw([]any{0, 1, 2, "a", false, 1.1})
+				return m
+			},
+			expectedLen:           3,
+			expectedNumericLabels: []float64{0, 1, 2}, // Explicitly check that zero is stored
+		},
+		{
+			desc: "drop non-float64 elements",
+			input: func() pcommon.Map {
+				m := pcommon.NewMap()
+				s := m.PutEmptySlice("k")
+				s.FromRaw([]any{0.0, 1.1, 1.2, 2, "a", true})
+				return m
+			},
+			expectedLen:           3,
+			expectedNumericLabels: []float64{0.0, 1.1, 1.2}, // Explicitly check that zero is stored
+		},
+		{
+			desc: "drop nil array values",
+			input: func() pcommon.Map {
+				m := pcommon.NewMap()
+				s := m.PutEmptySlice("k")
+				s.FromRaw([]any{nil, nil})
+				return m
+			},
+			expectedLen: 0,
+		},
+		{
+			desc: "ensure empty array does not panic",
+			input: func() pcommon.Map {
+				m := pcommon.NewMap()
+				s := m.PutEmptySlice("k")
+				s.FromRaw([]any{})
+				return m
+			},
+			expectedLen: 0,
+		},
+		{
+			desc: "ensure nill array does not panic",
+			input: func() pcommon.Map {
+				m := pcommon.NewMap()
+				s := m.PutEmptySlice("k")
+				s.FromRaw(nil)
+				return m
+			},
+			expectedLen: 0,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			e := &modelpb.APMEvent{
+				Event:         &modelpb.Event{},
+				Span:          &modelpb.Span{},
+				Labels:        make(map[string]*modelpb.LabelValue),
+				NumericLabels: make(map[string]*modelpb.NumericLabelValue),
+			}
+
+			otlp.TranslateSpan(ptrace.SpanKindInternal, tC.input(), e)
+
+			labels := e.GetLabels()[key].GetValues()
+			numericals := e.GetNumericLabels()[key].GetValues()
+			length := len(labels) + len(numericals)
+
+			assert.Equal(t, tC.expectedLen, length)
```

Comment:
```suggestion
			assert.Equal(t, tC.expectedLen, length)
```
this line is redundant given the 2 following assertions.